tipsy-bot/tipsy.py

```diff
@@ -1,66 +1,73 @@
+camera_name = os.getenv("ROBOT_CAMERA") or "cam"
+# change this if you named your sensors differently in your robor configuration
+sensor_names = (os.getenv("SENSOR_NAMES") or "ultrasonic,ultrasonic2").split(",")
+pause_interval = os.getenv("PAUSE_INTERVAL") or 3
+
+if isinstance(pause_interval, str):
+    pause_interval = int(pause_interval)
 
 base_state = "stopped"
 
+
 async def connect():
-    creds = Credentials(
-        type='robot-location-secret',
-        payload=robot_secret)
+    creds = Credentials(type="robot-location-secret", payload=robot_secret)
     opts = RobotClient.Options(
-        refresh_interval=0,
-        dial_options=DialOptions(credentials=creds)
+        refresh_interval=0, dial_options=DialOptions(credentials=creds)
     )
     return await RobotClient.at_address(robot_address, opts)
 
-async def obstacle_detect(sensor):
+
+async def obstacle_detect(sensor: Sensor):
     reading = (await sensor.get_readings())["distance"]
-    return reading 
+    return reading
+
+
+async def gather_obstacle_readings(sensors: list[Sensor]):
+    return await asyncio.gather(*[obstacle_detect(sensor) for sensor in sensors])
 
-async def obstacle_detect_loop(sensor, sensor2, base):
-    while(True):
-        reading = await obstacle_detect(sensor)
-        reading2 = await obstacle_detect(sensor2)
-        if reading < 0.4 or reading2 <0.4:
+
+async def obstacle_detect_loop(sensors: list[Sensor], base: Base):
+    while True:
+        distances = await gather_obstacle_readings(sensors)
+        if any(distance < 0.4 for distance in distances):
             # stop the base if moving straight
             if base_state == "straight":
                 await base.stop()
                 print("obstacle in front")
-        await asyncio.sleep(.01)
+        await asyncio.sleep(0.01)
+
 
-async def person_detect(detector, sensor, sensor2, base):
-    while(True):
+async def person_detect(detector: VisionClient, sensors: list[Sensor], base: Base):
+    while True:
         # look for person
         found = False
         global base_state
         print("will detect")
         detections = await detector.get_detections_from_camera(camera_name)
         for d in detections:
-            if d.confidence > .7:
+            if d.confidence > 0.7:
                 print(d.class_name)
-                if (d.class_name == "Person"):
+                if d.class_name == "Person":
                     found = True
-        if (found):
+        if found:
             print("I see a person")
             # first manually call obstacle_detect - don't even start moving if someone is in the way
-            distance = await obstacle_detect(sensor)
-            distance2 = await obstacle_detect(sensor2)
-            if (distance > .4 or distance2 > .4):
+            distances = await gather_obstacle_readings(sensors)
+            if all(distance > 0.4 for distance in distances):
```

Comment:
Using `all` here rather than `any` since we want to ensure none of the sensors are blocked before attempting to move forward.

Reply:
makes sense